app/code/AstoundDRudenko/Badge/Model/Product/View/Badge/Provider.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+namespace AstoundDRudenko\Badge\Model\Product\View\Badge;
+
+use AstoundDRudenko\Badge\Helper\Data as Helper;
+use AstoundDRudenko\Badge\Model\Attribute\Badge\Source;
+use Magento\Catalog\Model\Product;
+
+/**
+ * Model responsible for badge providing
+ *
+ * Class Provider
+ * @package AstoundDRudenko\Badge\Model\Product\View\Badge
+ */
+class Provider
+{
+    /**
+     * Default sort priority
+     */
+    public const DEFAULT_SORT_PRIORITY = 99999;
+
+    /**
+     * Sorted badges with priority
+     * @var array
+     */
+    private $configPriority;
+
+    /**
+     * Badges options
+     * @var array
+     */
+    private $badgeOptions;
+
+    /**
+     * @var Helper
+     */
+    private $helper;
+
+    /**
+     * @var Source
+     */
+    private $badgesSource;
+
+    /**
+     * @var array
+     */
+    private $badges = [];
+
+    /**
+     * @var bool
+     */
+    private $multipleBadges;
+
+    /**
+     * Badge constructor.
+     * @param Source $badgesSource
+     * @param Helper $helper
+     */
+    public function __construct(
+        Source $badgesSource,
+        Helper $helper
+    ) {
+        $this->helper = $helper;
+        $this->badgesSource = $badgesSource;
+        $this->initConfiguration();
+    }
+
+    /**
+     * Is badges enabled
+     * @return bool
+     */
+    public function isBadgesEnabled()
+    {
+        return (bool)$this->helper->isBadgesEnabled();
```

Comment:
your config model should already return ONLY bool values, no need to cast to bool here